doc/sphinxman/source/add_tests.rst

```diff
@@ -30,108 +30,312 @@
+
+* For a CMS user, a test suite alongside installed |PSIfour| can be used to show that this piece is working in a complex software environment.
+
+* For a feature developer, adding tests provides confidence that you
+  can leave your code untouched and still advertise that the feature works
+  years later. With tests, proposed changes to |PSIfour| that break your
+  code fall upon the change proposer to fix, rather than being merged
+  silently and lying in wait for a concientious user to detect and report
+  and then likely falling upon *you* to fix.
+
+* For a general developer, the test suite allows confidence in refactoring, switching out underlying libraries, maintenance, and upgrading.
+
+
+CTest and pytest, PSIthon and PsiAPI
+====================================
+
+In designing a test, sometimes you want it to be a model input for the user in a single file or you don't want a lot of ``psi4.`` or Python syntax cluttering the input.
+In this case, follow :ref:`faq:add_psithon_tests` to prepare as PSIthon (``psi4 input.dat``) for, roughly speaking, running through :program:`ctest`.
+The PSIthon/CTest test suite occupies the whole of :source:`tests` *except* :source:`tests/pytests`.
+
+At other times you want the test to check several variations of a template job or you want to test error handling or you want to focus on PsiAPI rather than PSIthon or you want to control the compute conditions with environment variables.
+In this case, follow :ref:`faq:add_psiapi_tests` to prepare as PsiAPI (``import psi4``) for, roughly speaking, running through :program:`pytest`.
+The PsiAPI/pytest test suite occupies :source:`tests/pytests`.
+
+The above description sounds as if there are two disjoint test suites, and you have to run both ``ctest`` and ``pytest`` to fully test |PSIfour|.
+This has indeed been the case until March 2022.
+The difficulty has been that (1) two test suites is unexpected so some developers don't know to run both; and (2) there are important tests in the PSIthon suite that can't be run on a |PSIfour| installation since CTest only works in a build directory.
+Now, by adding an extra file to the test directory, PSIthon tests can also be run through :program:`pytest`.
+This hasn't rolled out to all ~500 PSIthon tests (help wanted), but eventually |PSIfour| can be tested with a single command from a build or from an installation.
+Therefore, in designing a test, choose its mode based on whether PSIthon or PsiAPI suits it better and whether it's a simple model for users (probably PSIthon) or for expert users (probably PsiAPI).
+Both will continue to work in future, and neither have limitations.
+
+
+Test Contents
+=============
+
+* If you're writing a new class or closely examining an old one and you develop unit tests for probing its behavior, by all means add them to the test suite.
+  But on the whole, |PSIfour| tests should be integration tests focusing on non-regression of user input to answers.
```

Comment:
```suggestion
* Most tests will be |PSIfour| tests integration tests focusing on non-regression of user input to answers, and we insist on having these.
  But if you find unit tests helpful, by all means add them to the test suite.
```

Put the important part first, remove the assumption about unit testing being for a class, and clarify what we demand for tests vs. what we predict for tests.